### PR TITLE
Retry listing artifacts if list is empty

### PR DESCRIPTION
**Pull request by ynouri**
This is a draft attempt at solving #43. Tests are not passing yet.

Idea is to use the p-retry package:
https://www.npmjs.com/package/p-retry

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -4,8 +4,36 @@ const AdmZip = require('adm-zip')
...
+        artifacts = artifacts.data.artifacts
+    }
+
+    if (artifacts.length == 0) {
+        console.log("No artifacts found yet.");
+        return Promise.reject("No artifacts found.");
+    }
+
+    return artifacts;
+}
```
Comment (ynouri) [inline index.js:33]: We put the artifacts listing into its own async function. If no artifacts are found, it rejects the promise.

File: index.js [hunk 2]
```diff
@@ -84,23 +112,10 @@ async function main() {
...
-        } else {
-            artifacts = artifacts.data.artifacts
-        }
-
-        if (artifacts.length == 0)
-            throw new Error("no artifacts found")
+        let artifacts = await pRetry(
+        () => listArtifacts(client, owner, repo, runID, name),
+        {retries: 5}
+        );
```
Comment (ynouri) [inline index.js:118]: The async `listArtifacts` function can then be used inside `pRetry`

**Comment by ynouri** [comment]
Note that as I was adding more dependencies for retry and for unit testing, the `node_modules` folder was becoming really huge, hence I thought it could be cleaner to move to the ncc-based solution described in the [Github Documentation](https://docs.github.com/en/actions/creating-actions/creating-a-javascript-action#commit-tag-and-push-your-action-to-github).

The distribution can be built by running:
```
ncc build index.js --license LICENSE
```

This removes the need to version control `node_modules`. But the main caveat is that the distribution needs to be built manually on each PR.

**Comment by dawidd6** [comment]
You definitely packed too much changes in this PR. I want to keep using `node_modules`, keep the main function named `main`, etc. Please just implement the feature, nothing more.

**Comment by kkalugerov** [comment]
Hi guys, are you going to merge this ?

**Comment by dawidd6** [comment]
See my previous comment on this.

**Comment by kkalugerov** [comment]
Hi @ynouri, are you planning to finish this PR in the way @dawidd6 asked you ? :)

**Comment by ynouri** [comment]
Hey @kkalugerov , due to lack of time and changes of incentives (the need for retrieving artifacts my team's workflows recently disappeared) it will be difficult for me to push this to the finish line in a short horizon. I will close the PR, but feel free to take inspiration from this first attempt - felt like the `p-retry` approach was promising

**Pull request closed without merging**